sm/download/src/download.rs

```diff
@@ -72,15 +91,19 @@ impl Downloader {
         })
         .await?;
 
-        let content = response.bytes().await?;
+        let len: u64 = match response.content_length() {
+            Some(package_size) => package_size,
+            None => self.reserved_buffer_size as u64,
```

Comment:
I understand this is a workaround for the situation where the server doesn't send the length header?

Reply:
Yes, thats right